src/@utils/api.ts

```diff
@@ -1,291 +1,279 @@
+import axios from "axios";
+import Cookies from "js-cookie";
 
 import Config from "../@config";
-import {
+
+import type { Repo, Space, User } from "../@types/user";
+import type {
   CreateProjectOptions,
-  Project,
-  ProjectId,
-  Repo,
   Response,
-  Space,
-  User,
-  SchemaData,
-  UpdateProjectOption,
-  UpdateProjectOptions,
-  UpdateProjectBuildOption,
+  AddProjectOptions,
   UpdateProjectBuildOptions,
   DeleteProjectOption,
+  DeleteProjectOptions,
 } from "../@types/api";
+import type { Project, ProjectId } from "../@types/project";
+import type { SchemaData, Webhook } from "../@types/cms";
+
+const client = axios.create({
+  withCredentials: true,
+  baseURL: Config.SERVER_URL_API,
+  params: {},
+  timeout: 2500,
+});
+
+export function logout() {
+  Cookies.remove("userId");
+  Cookies.remove("loginData");
+  Cookies.remove("accessToken");
+  Cookies.remove("githubAccessToken");
+}
 
-class APIClient {
-  private accessToken: string = "";
-  private githubAccessToken: string = "";
-  private userId: string = "";
-
-  private client = (): AxiosInstance => {
-    const client = axios.create({
-      baseURL: Config.SERVER_URL_API,
-      headers: {
-        Authorization: `Bearer ${this.accessToken}`,
-      },
-      params: {},
-      timeout: 2500,
-    });
-
-    client.interceptors.request.use(config => {
-      config.params = {
-        githubAccessToken: this.githubAccessToken,
-        ...config.params,
-      };
-      return config;
-    });
-
-    return client;
-  };
-
-  setAccessToken(accessToken?: string): APIClient {
-    this.accessToken = accessToken ?? "";
-    return this;
-  }
-
-  setGithubAccessToken(githubAccessToken?: string): APIClient {
-    this.githubAccessToken = githubAccessToken ?? "";
-    return this;
-  }
+export async function getSpaces(): Promise<Space[]> {
+  try {
+    const { data } = await client.get<Response<Space[]>>(`/user/spaces`);
 
-  setUserId(userId?: string): APIClient {
-    this.userId = userId ?? "";
-    return this;
+    return data.result;
+  } catch (error) {
+    throw error;
   }
+}
 
-  async getSpaces(): Promise<Space[]> {
-    try {
-      const { data } = await this.client().get<Response<Space[]>>(
-        `/users/${this.userId}/spaces`,
-      );
+export async function getSpaceRepos(space: Space): Promise<Repo[]> {
+  try {
+    const { data } = await client.get<Response<Repo[]>>(
+      `/user/spaces/${space.installId}/repos`,
+    );
 
-      return data.result;
-    } catch (error) {
-      throw error;
-    }
+    return data.result;
+  } catch (error) {
+    throw error;
   }
+}
 
-  async getSpaceRepos(space: Space): Promise<Repo[]> {
-    try {
-      const { data } = await this.client().get<Response<Repo[]>>(
-        `/users/${this.userId}/spaces/${space.installId}/repos`,
-      );
+export async function getUserData(): Promise<User> {
+  try {
+    const { data } = await client.get(`/user`);
```

Comment:
```suggestion
    const { data } = await client.get<Response<User>>(`/user`);
```

Generic을 넣으면 더 Type-safe해질 것 같습니다~